### PR TITLE
loadbalancer: Change over to OVSDB bindings.

### PR DESCRIPTION
**Pull request by shettyg**

### CODE REVIEW

**Comment by shettyg** [comment]
@TomCodeLV   

Can you please glance this through to see if the work looks sane?

File: go-controller/pkg/ovn/loadbalancer.go [hunk 1]
```diff
@@ -1,108 +1,152 @@
...
+	var defaultGateway string
+
+	gateways := ovn.ovnNbCache.GetMap("Logical_Router", "uuid")
+	for _, gateway := range gateways {
+		options := gateway.(map[string]interface{})["options"]
+		if options == nil {
+			continue
+		}
+		option := options.(map[string]interface{})["lb_force_snat_ip"]
+		if option != nil && option.(string) == "100.64.1.2" {
```
Comment (TomCodeLV) [inline go-controller/pkg/ovn/loadbalancer.go]: "100.64.1.2" this looks like a magic one thing, I would suggest to store this IP in meaningfully named variable or at least write a comment what is this IP address representing

File: go-controller/pkg/ovn/loadbalancer.go [hunk 2]
```diff
@@ -1,108 +1,152 @@
...
+	return ""
 }
 
 func (ovn *Controller) getLoadBalancerVIPS(
 	loadBalancer string) (map[string]interface{}, error) {
-	outStr, _, err := util.RunOVNNbctlHA("--data=bare", "--no-heading",
-		"get", "load_balancer", loadBalancer, "vips")
-	if err != nil {
-		return nil, err
+	lb := ovn.ovnNbCache.GetMap("Load_Balancer", "uuid", loadBalancer)
```
Comment (TomCodeLV) [inline go-controller/pkg/ovn/loadbalancer.go]: I would suggest to rename loadBalancer to loadBalancerId to explicitly show that it is an Id and not a whole object

File: go-controller/pkg/ovn/ovn.go [hunk 3]
```diff
@@ -118,15 +118,18 @@ func dialNB() (*ovsdb.OVSDB, *dbcache.Cache) {
...
+				"Port_Group":     {"_uuid", "name", "acls", "ports"},
+				"Load_Balancer":  {"_uuid", "vips", "protocol", "external_ids"},
+				"Logical_Router": {"_uuid", "name", "options"},
 			},
 			Indexes: map[string][]string{
 				"Logical_Switch_Port": {"name"},
 				"Logical_Switch":      {"name"},
 				"Address_Set":         {"name"},
 				"Port_Group":          {"name"},
+				"Logical_Router":      {"name"},
```
Comment (TomCodeLV) [inline go-controller/pkg/ovn/ovn.go]: If I am not mistaking this index is not used.

**Comment by shettyg** [comment]
Thanks. I removed the index for Logical_Router. I let the other changes remain as they are part of original code and I do not want to complicate the merge process back to master.

**Pull request merged**